pulsar/impl_message.go

```diff
@@ -126,3 +142,36 @@ func (msg *message) EventTime() time.Time {
 func (msg *message) Key() string {
 	return msg.key
 }
+
+func newAckTracker(size int) *ackTracker {
```

Comment:
Since this ack tracker is self-contained, should we move it into a separate file, perhaps in the `internal` pkg?

Reply:
I think it makes sense in this file since it's related to the message id implementation.